### PR TITLE
Dunno why these commits were removed

### PR DESCRIPTION
**Pull request by killzoms**
Added server command processor to process commands typed into server console

### CODE REVIEW

File: NitroxServer/Program.cs [hunk 1]
```diff
@@ -26,7 +27,7 @@ static void Main(string[] args)
 
             while (true)
```
Comment (Measurity) [inline NitroxServer/Program.cs]: If you add a `Server.IsRunning` to the `Server` class you can use is here to quit reading commands and exit/close the console. Then you can write a close command and shutdown the server.

File: NitroxServer/ConsoleCommands/Processor/ConsoleCommandProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            IEnumerable<Command> discoveredCommands = Assembly.GetExecutingAssembly()
+                .GetTypes()
+                .Where(p => typeof(Command).IsAssignableFrom(p) &&
+                            p.IsClass && !p.IsAbstract
+                      )
+                .Select(Activator.CreateInstance)
+                .Cast<Command>();
+            foreach (Command cmd in discoveredCommands)
+            {
+                cmd.Init();
```
[the remark this answers is not in the document]
Comment (killzoms) [reply NitroxServer/ConsoleCommands/Processor/ConsoleCommandProcessor.cs]: I havent the foggiest on how to implement the constructor hence use of Init method

File: NitroxServer/Program.cs (hunk 1 above)
Comment (killzoms) [reply NitroxServer/Program.cs]: Closing server from this while loop may cause more issues than closing from the UdpServer thread

File: NitroxServer/ConsoleCommands/Processor/ConsoleCommandProcessor.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (killzoms) [reply NitroxServer/ConsoleCommands/Processor/ConsoleCommandProcessor.cs]: Figured it out

File: NitroxServer/ConsoleCommands/SaveCommand.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        {
+            Name = "save";
+        }
+
+        public override void RunCommand(string[] args)
+        {
+            Server.Instance.Save();
+        }
+
+        public override string[] Args()
```
[the remark this answers is not in the document]
Comment (killzoms) [reply NitroxServer/ConsoleCommands/SaveCommand.cs]: its a property now

File: NitroxServer/ConsoleCommands/Processor/ConsoleCommandProcessor.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        {
+            if (msg.IsEmpty())
+            {
+                return;
+            }
+
+            Command cmd;
+            string[] args = msg.Split(' ');
+
+            if (commands.TryGetValue(args[0], out cmd))
```
[the remark this answers is not in the document]
Comment (Measurity) [reply NitroxServer/ConsoleCommands/Processor/ConsoleCommandProcessor.cs]: Haha, I could have done better. I meant it like this: 
```CS
args = text.Split()
.Select(arg => arg.Trim())
.Where(arg => !string.IsNullOrEmpty(arg))
```

File: NitroxServer/Program.cs (hunk 1 above)
Comment (Measurity) [reply NitroxServer/Program.cs]: It should wait for UdpServer to exit.

1. Wait for UdpServer to exit.
2. Wait for Server to exit.
3. Exit while loops in Program.cs and let it exit itself.

**Pull request merged**